### PR TITLE
Setting MaxUnavailable to 1 when calculation against # of available calico node pods resolves to 0

### PR DESCRIPTION
**Pull request by freecaykes**
## Description

Due to the potential rounding down of numNodesMaxUnavailable = (csMaxUnavailable %) * ksD+csD, where maxUnavailable is a percentage value, ,it may resolve to zero. Then we should default back maxUnavailable to 1 on the theory that surge might not work due to quota.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/controller/migration/namespace_migration.go
Comment (freecaykes) [inline pkg/controller/migration/namespace_migration.go:590]: can we set this to using int ? or should we set the windows one to using int32 ?

Comment (caseydavenport) [reply pkg/controller/migration/namespace_migration.go:590]: Any API fields should not use `int` and should instead use either `int32` or `int64` per the API conventions, so eventually this needs to be converted to an in32 (since that's the type of the maxunavailable field in the API)

> All public integer fields MUST use the Go (u)int32 or Go (u)int64 types, not (u)int (which is ambiguous depending on target platform). Internal types may use (u)int.

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md

Comment (freecaykes) [reply pkg/controller/migration/namespace_migration.go:590]: changing the windows on to int32 https://github.com/tigera/operator/pull/1804/commits/6882da130e88f4cba96d1f1738e44ac45cb41414

**Review by tmjd — CHANGES_REQUESTED** [review]
The K8s code that we're adjusting our calculation to match, does it log if maxUnavailable evaluates to zero (and maxSurge is zero)?
I can just imagine getting a report at some point about this being an error when it is just because they have too few nodes and hit the maxUnavailable 0 case. I think we should separate the cases to log error if err is returned but otherwise it is just an info level message, then I think the message can be more descriptive of the actual issue.

File: pkg/controller/installation/windows/calico_windows_upgrader.go
Comment (tmjd) [inline pkg/controller/installation/windows/calico_windows_upgrader.go:44]: can't this be defaulted to be an int32 with `defaultMaxUnavailable int32 = 1`, then the cast later isn't needed

Comment (freecaykes) [reply pkg/controller/installation/windows/calico_windows_upgrader.go]: https://github.com/tigera/operator/pull/1804/commits/f940907e37306e04655304bd9d771cc0b96bc1fb

**Review by tmjd — APPROVED** [review]
LGTM

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
I'm sorry but I failed delete the branch after merging the pull request.

**Pull request merged**
